common/logisticspipes/gui/guidebook/GuiGuideBookSlider.java

```diff
@@ -0,0 +1,87 @@
+package logisticspipes.gui.guidebook;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.audio.SoundHandler;
+import net.minecraft.client.gui.GuiButton;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.MathHelper;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import logisticspipes.LPConstants;
+
+public class GuiGuideBookSlider extends GuiButton {
+
+	private static final ResourceLocation GUI_BOOK_TEXTURE = new ResourceLocation(LPConstants.LP_MOD_ID, "textures/gui/guide_book.png");
+
+	@Getter
+	private float progress;
+	private boolean dragging;
+	public int max, min;
+
+	public void setProgress(float progress){
+		this.y = (int)((this.max-this.min)*progress);
+	}
+
+	public GuiGuideBookSlider(int buttonId, int x, int yTop, int yBot, float z, float progress, int widthIn, int heightIn) {
+		super(buttonId, x, yTop, widthIn, heightIn, "");
+		this.progress = progress;
+		this.enabled = true;
+		this.zLevel = z;
+		this.max = yTop;
+		this.min = yBot - heightIn;
+		this.y = (int) (max + (min - max) * progress);
+	}
+
+	public void reset(){
+		this.y = max;
+		this.progress = 0.0F;
+	}
+
+	@Override
+	public void drawButton(Minecraft mc, int mouseX, int mouseY, float partialTicks) {
+		if (this.visible)
+		{
+			mc.getTextureManager().bindTexture(GUI_BOOK_TEXTURE);
+			GlStateManager.color(1.0F, 1.0F, 1.0F, 1.0F);
+			this.hovered = mouseX >= this.x && mouseY >= this.y && mouseX < this.x + this.width && mouseY < this.y + this.height;
+			int i = this.hovered&&!dragging||!this.enabled?1:0;
+			int j = this.dragging||!this.enabled?1:0;
```

Comment:
Replace these with booleans with readable names, and then use the `? :` in the line below to make it more readable

Reply:
Yes right I was hacking away at the time sry about that

Reply:
Not sure if what I did was exactly what you meant but I'm sure it helps readability